modules/light-clients/08-wasm/types/msgs.go

```diff
@@ -33,3 +35,70 @@ func (m MsgStoreCode) GetSigners() []sdk.AccAddress {
 	}
 	return []sdk.AccAddress{signer}
 }
+
+type QueryMsg struct {
+	Status               *statusMsg               `json:"status,omitempty"`
+	ExportMetadata       *exportMetadataMsg       `json:"export_metadata,omitempty"`
+	TimestampAtHeight    *timestampAtHeightMsg    `json:"timestamp_at_height,omitempty"`
+	VerifyClientMessage  *verifyClientMessageMsg  `json:"verify_client_message,omitempty"`
+	VerifyMembership     *verifyMembershipMsg     `json:"verify_membership,omitempty"`
+	VerifyNonMembership  *verifyNonMembershipMsg  `json:"verify_non_membership,omitempty"`
+	CheckForMisbehaviour *checkForMisbehaviourMsg `json:"check_for_misbehaviour,omitempty"`
+}
+
+type (
+	statusMsg            struct{}
+	exportMetadataMsg    struct{}
+	timestampAtHeightMsg struct {
+		Height exported.Height `json:"height"`
+	}
+	verifyClientMessageMsg struct {
+		ClientMessage exported.ClientMessage `json:"client_message"`
+	}
+	verifyMembershipMsg struct {
+		Height           exported.Height `json:"height"`
+		DelayTimePeriod  uint64          `json:"delay_time_period"`
+		DelayBlockPeriod uint64          `json:"delay_block_period"`
+		Proof            []byte          `json:"proof"`
+		Path             exported.Path   `json:"path"`
+		Value            []byte          `json:"value"`
+	}
+	verifyNonMembershipMsg struct {
+		Height           exported.Height `json:"height"`
+		DelayTimePeriod  uint64          `json:"delay_time_period"`
+		DelayBlockPeriod uint64          `json:"delay_block_period"`
+		Proof            []byte          `json:"proof"`
+		Path             exported.Path   `json:"path"`
+	}
+	checkForMisbehaviourMsg struct {
+		ClientMessage exported.ClientMessage `json:"client_message"`
```

Comment:
Instead of having two fields (`Header` and `Misbehaviour`) we pass directly `ClientMessage`, as we discussed during our own internal walkthrough. I think I mentioned this already to Composable, but I will double check.